client/my-sites/customize/main.jsx

```diff
@@ -69,8 +70,13 @@ var Customize = React.createClass( {
 	},
 
 	redirectIfNeeded: function( menusUrl, site ) {
+		if ( isRedirecting ) {
+			return;
+		}
 		if ( site && menusUrl !== '/customize/menus/' + site.slug ) {
```

Comment:
We really should compare `menusUrl` to the current path, which we can fetch from `context`...